subsys/bluetooth/host/hci_core.c

```diff
@@ -2936,6 +2969,25 @@ static void hci_encrypt_key_refresh_complete(struct net_buf *buf)
 }
 #endif /* CONFIG_BT_SMP || CONFIG_BT_BREDR */
 
+#if defined(CONFIG_BT_CONN)
+static void bt_hci_evt_read_remote_version_complete(struct net_buf *buf)
+{
+	struct bt_hci_evt_remote_version_info *evt;
+	struct bt_conn_rv_info rv;
+
+	evt = net_buf_pull_mem(buf, sizeof(*evt));
+
+	if (!evt->status) {
+		rv.version = evt->version;
+		rv.manufacturer = sys_le16_to_cpu(evt->manufacturer);
+		rv.subversion = sys_le16_to_cpu(evt->subversion);
+	}
+	remote_version_cb(rv);
```

Comment:
If evt->status is non-zero here how can the user know? I think it would be best to give a pointer to remote version, then we can give NULL if the procedure failed.

Reply:
I have now changed to pointers (not yet pushed), should I add a status field to the bt_conn_rv_info so that if there is an error the developer can know the error code?
Or should I just stick with giving NULL?